lib/pbench/server/api/resources/query_apis/datasets/table_of_content.py

```diff
@@ -38,49 +41,85 @@ def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
         that belong to the given run id.
 
         Args:
-            json_data: JSON dictionary of type-normalized parameters
+            json_data: JSON dictionary of type-normalized parameters,
+                    we pull 'parent' key value from it.
             context: propagate the requested set of metadata to the
-                    postprocess method.
-                "run_id": Dataset document ID
+                    postprocess method, it also inludes "run_id" which
+                    is Dataset document ID
 
         EXAMPLE:
-        We read "run_id" from 'context'
         {
             "run_id": "1234567890"
-            "parent": '/default'
+            "parent": '/1-default'
         }
         """
-        parent = json_data.get("parent")
+        self.parent = json_data.get("parent")
```

Comment:
Good point. That's why the `CONTEXT` concept exists; as Webb says, `self` isn't specific to the operation we're performing.

Also, setting this in `assemble` doesn't make sense unless you're using it in `postprocess`, which you don't seem to be doing.

Although that gets confusing, as you seem to have *two* `assemble` methods in this class, at line 36 and again at line 238 ... with a second `postprocess` following it. ???

I'm skimming through here today because Webb replied, and I was curious ... however after seeing that you hadn't asked for re-review, and with your hardcoded index, I'd assumed you weren't ready for review and maybe all this will settle out before you are.

Reply:
I tried adding it in context but then `get_index` function was throwing some tantrums. I will check again :)

Reply:
Riya, my apologies if I jumped in and started reviewing too soon!

BTW, Dave, I'm not sure which code you're looking at -- I only see _one_ `assemble` method (the module ends shortly before line 238...), and I _did_ see the parent value being used in `postprocess`...so I suspect that we're looking at different revisions.